Assets/Plugins/CountlySDK/Services/LocationService.cs

```diff
@@ -0,0 +1,158 @@
+                IPAddress = null;
+                CountryCode = null;
+                IsLocationDisabled = countlyConfiguration.IsLocationDisabled;
+            }
+            else
+            {
+                City = countlyConfiguration.City;
+                Location = countlyConfiguration.Location;
+                IPAddress = countlyConfiguration.IPAddress;
+                CountryCode = countlyConfiguration.CountryCode;
+                IsLocationDisabled = countlyConfiguration.IsLocationDisabled;
+            }
+        }
+
+        internal async Task SendRequestWithEmptyLocation()
+        {
+            var requestParams =
+               new Dictionary<string, object>();
+
+
+            requestParams.Add("metrics", JsonConvert.SerializeObject(CountlyMetricModel.Metrics, Formatting.Indented,
+                new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore }));
+
+            requestParams.Add("location", string.Empty);
+
+            await _requestCountlyHelper.GetResponseAsync(requestParams);
+        }
+
+        internal async Task SendIndependantLocationRequest()
+        {
+
+            if (!_consentService.CheckConsent(Features.Location))
+            {
+                return;
+            }
+            
+            var requestParams =
+                new Dictionary<string, object>();
+
+            /*
+             * Empty country code, city and IP address can not be sent.
+             */
+
+            if (!string.IsNullOrEmpty(IPAddress))
+            {
+                requestParams.Add("ip_address", IPAddress);
+            }
+
+            if (!string.IsNullOrEmpty(CountryCode))
+            {
+                requestParams.Add("country_code", CountryCode);
+            }
+
+            if (!string.IsNullOrEmpty(City))
+            {
+                requestParams.Add("city", City);
+            }
+
+            if (!string.IsNullOrEmpty(Location))
+            {
+                requestParams.Add("location", Location);
+            }
+
+            await _requestCountlyHelper.GetResponseAsync(requestParams);
+
+        }
+
+
+
+        /// <summary>
+        /// Disabled the location tracking on the Countly server
+        /// </summary>
+        public async void DisableLocation()
+        {
+            IsLocationDisabled = true;
+            City = null;
+            IPAddress = null;
+            CountryCode = null;
+            Location = string.Empty;
+
+            /*
+             *If the location feature gets disabled or location consent is removed,
+             *the SDK sends a request with an empty "location". 
+             *TODO ~ On Consent removed
+             */
+
+            await SendRequestWithEmptyLocation();
+        }
+
+        /// <summary>
+        /// Set Country code (ISO Country code), City, Location and IP address to be used for future requests.
+        /// </summary>
+        /// <param name="countryCode">Country code (ISO Country code)</param>
```

Comment:
You haven't really provided any information of what the expected data format is. For any of these 4 params